### DISCUSSION TITLE
Adding interactive viewer and report to pyQt Widget

### DISCUSSION THREAD

**Discussion by johanna-LSI** [KarchinLab/open-cravat#73] [Q&A] [answered]
Hello everybody,
I have just come across open-cravat and think it is a great tool, which is why I want to try to integrate it in my python desktop application written in pyQt. Do you think this is possible? If so, could you point me into the right direction to add the report/interactive viewer to a pyQt Widget?

Thanks in Advance!
Regards,
Johanna

**Comment by rkimoakbioinformatics** [comment, marked as answer]
Hi Johanna,

We don't have detailed documentation on it yet, but basically you can import `cravat` package in your Python code and extract result data using it. For example,

```
import asyncio
import cravat

def ex(f):
    ret = loop.run_until_complete(f)
    return ret

loop = asyncio.new_event_loop()
cf = ex(cravat.cravat_filter.CravatFilter.create(dbpath="example_input.sqlite", mode="sub"))
datarows = [row for row in ex(cf.exec_db(cf.get_filtered_iterator, "variant"))]
print(datarows[1][0])
ex(cf.close_db())
```

We are planning to release a tutorial on such a use case as yours in the future. Good luck with your endeavor and let us know how it goes and if you have further questions.
